activation/handler.go

```diff
@@ -385,10 +384,8 @@ func (h *Handler) storeAtx(ctx context.Context, atx *types.VerifiedActivationTx)
 						Data: &atxProof,
 					},
 				}
-				if err := malfeasance.ValidateAndSave(ctx, h.log, h.cdb, dbtx, h.edVerifier, nil, &types.MalfeasanceGossip{
-					MalfeasanceProof: *proof,
-				}); err != nil && !errors.Is(err, malfeasance.ErrKnownProof) {
-					return fmt.Errorf("adding malfeasance proof: %w", err)
+				if err := h.cdb.AddMalfeasanceProof(atx.SmesherID, proof, dbtx); err != nil {
```

Comment:
this api call is still a risk. one thing is that it is confusing, as cache has access to its own db pointer. 
but also there is mutex on this codepath, if this mutex will be abused then every write will be delayed. this kind of bugs are extremely hard to find and debug later

i personally would avoid using that directly, even at inconvenience that proof won't be cached immediately

Reply:
thanks for the suggestion. i agree.

the tricky thing is honest nodes' maliciousness is also cached (so we don't need to hit db every time a (hare) msg is received). so it's important to keep the db and cache in sync. 

i replaced `AddMalfeasanceProof` with `CacheMalfeasanceProof`.
now the code is changed to saving the proof directly to db, and then call `cdb.CacheMalfeasanceProof()` when the db transaction is successfully terminated.